### PR TITLE
Handle both async and non async Alma client with the same endpoint code

### PR DESCRIPTION
**Pull request by Natim**

### CODE REVIEW

File: alma/endpoints/payments.py [hunk 1]
```diff
@@ -10,14 +10,13 @@ class Payments(Base):
     PAYMENTS_PATH = "/v1/payments"
 
     def eligibility(self, order_data):
-        response = (
+        return (
```
Comment (Natim) [inline alma/endpoints/payments.py:12]: I need to do that for all endpoints now.

File: README.md [hunk 2]
```diff
@@ -39,3 +43,34 @@ eligibility = alma_client.payments.eligibility(payment_data)
...
+
+alma_client = AsyncClient.with_api_key("sk_test..")
+payments = await alma_client.payments.fetch_all()
+
+for p in payments:
+    print(f"{p.id}: Paiement en {len(p.payment_plan)} fois")
+
+
+payment_data = {
+    "purchase_amount": 10000,
```
Comment (olance) [inline README.md]: Sorry I just updated `README.md` on `main` with a corrected version of that code, which misses the wrapping `"payment"` object. Can you rebase?

File: alma/client.py [hunk 3]
```diff
@@ -102,14 +162,12 @@ def __init__(
...
-                "`mode` option must be one of ({LIVE}, {TEST})".format(
-                    LIVE=ApiModes.LIVE.value, TEST=ApiModes.TEST.value
-                )
+                f"`mode` option must be one of ({ApiModes.LIVE.value}, {ApiModes.TEST.value})"
             )
 
         self.context = Context(options)
         self.init_user_agent()
-        self._endpoints = {}  # type: ignore
+        self._endpoints = {}  # type: Dict[str, endpoints.Base]
```
Comment (olance) [inline alma/client.py]: If we drop 3.5 support then we should be able to do this:

```suggestion
        self._endpoints: Dict[str, endpoints.Base] = {}
```

File: alma/endpoints/payments.py [hunk 4]
```diff
@@ -2,25 +2,22 @@
...
     PAYMENTS_PATH = "/v1/payments"
 
     def eligibility(self, order_data):
-        response = (
-            self.request("{PAYMENTS_PATH}/eligibility".format(PAYMENTS_PATH=self.PAYMENTS_PATH))
+        return (
+            self.request(f"{self.PAYMENTS_PATH}/eligibility")
             .set_body(order_data)
             .post()
+            .expect(lambda response: Eligibility(response.json))
```
Comment (olance) [inline alma/endpoints/payments.py]: Can't we use `expectJson(Eligibility)` here?
```suggestion
            .expectJson(Eligibility)
```

**Review by olance — APPROVED** [review]

**Review by olance — APPROVED** [review]

**Pull request merged**